Makefile

```diff
@@ -24,20 +24,28 @@ deps:
 checks:
 	./scripts/checks "$(PKGS)"
 
-build: deps checks
+run-build: deps checks
 	godep go install -v $(TO_BUILD)
 
+build: start
+	vagrant ssh netplugin-node1 -c 'sudo -i bash -lc "source /etc/profile.d/envvar.sh && cd /opt/golang/src/github.com/contiv/netplugin && make run-build"'
+
 clean: deps
 	rm -rf Godeps/_workspace/pkg
 	godep go clean -i -v ./...
 
 # setting CONTIV_NODES=<number> while calling 'make demo' can be used to bring
 # up a cluster of <number> nodes. By default <number> = 1
-demo: build
-	CONTIV_HOST_GOBIN=$(HOST_GOBIN) CONTIV_HOST_GOROOT=$(HOST_GOROOT) vagrant up
+start: 
+	CONTIV_NODE_OS=${CONTIV_NODE_OS} vagrant up
+
+demo-centos:
+	CONTIV_NODE_OS=centos make demo
 
-clean-demo:
-	vagrant destroy -f
+stop:
+	CONTIV_NODES=2 vagrant destroy -f
+
+demo: stop start build
```

Comment:
at least for the testing tasks, using `stop` here is absolutely essential. if the binaries are not built on the right VM (e.g., ubuntu vs. centos), the whole thing explodes.

I can remove it from demo, but I figured if someone types `make demo` that they probably want a new set of boxes.